File: python/Problem326.py
# Problem: 326. Power of Three
# URL: https://leetcode.com/problems/power-of-three/
# Difficulty: Easy
# Tags: Math

class Solution:
  def isPowerOfThree(self, n: int) -> bool:
    if n <= 0:
      return False
    while n % 3 == 0:
      n /= 3
    return n == 1

File: cpp/Problem326.cpp
/*
 * Problem: 326. Power of Three
 * URL: https://leetcode.com/problems/power-of-three/
 * Difficulty: Easy
 * Tags: Math
 */

class Solution {
public:
	bool isPowerOfThree(int n) {
		if (n <= 0) {
			return false;
		}
		while (n % 3 == 0) {
			n /= 3;
		}
		return n == 1;
	}
};